test/Spec/ServerSpec.hs

```diff
@@ -35,5 +43,19 @@ serverTestsIO =
         in get "hello.html" `shouldRespondWith` expectedResp
       it "dir/inner-file.html responds correctly" $
         get "dir/inner-file.html" `shouldRespondWith` "Inner File\n"
+      it "dir/test-gzipped.js responds correctly with JS script" $ do
+        gzippedJS <- liftIO . ByteString.readFile $ testDir </> "dir" </> "test-gzipped.js.gz"
+        let response = ResponseMatcher { matchHeaders = ["Content-Encoding" <:> "gzip"]
+                                       , matchBody    = bodyEquals gzippedJS
+                                       , matchStatus  = 200
+                                       }
+        get "dir/test-gzipped.js" `shouldRespondWith` response
+      it "dir/test-brotli.js responds correctly with JS script" $ do
+        brotliJS <- liftIO . ByteString.readFile $ testDir </> "dir" </> "test-brotli.js.br"
+        let response = ResponseMatcher { matchHeaders = ["Content-Encoding" <:> "br"]
+                                       , matchBody    = bodyEquals brotliJS
+                                       , matchStatus  = 200
+                                       }
+        get "dir/test-brotli.js" `shouldRespondWith` response
```

Comment:
Could you add one more test here for a gzipped html file? (This isn't super high priority, but would be nice since the code paths for html and other file types are slightly different...)

Reply:
updated